dracut/30ignition/persist-ifcfg.sh

```diff
@@ -33,6 +33,16 @@ persist_ifcfg() {
         return 0
     fi
 
+    # Only persist the hostname if something else has not populated
+    # /etc/hostname, such as Ignition Files.
+    hname="$(< /proc/sys/kernel/hostname)"
+    hpath="/sysroot/etc/hostname"
+    if [ ! -e "${hpath}" ] && [ -n "${hname}" ]; then
+        echo -ne "${hname}" > "${hpath}"
```

Comment:
The `-e` isn't necessary here right?

It also looks to me like at least systemd does emit a newline character for the hostname - generally friendlier to shells and scripts to have a trailing newline.

So let's drop the `-ne`?

Reply:
I added `-ne` for two reasons:
1. `systemd` does include the line, but other tooling does not. For example: ```
podman run -it --volume=/:/host:ro --cap-drop=AUDIT_WRITE,MKNOD,NET_RAW --env=TERM=xterm-256color --rm=true bash
bash-5.0# cat /etc/hostname
6b5dc518a2a0bash-5.0#
```
2. Style: I cargo culted it.